modules/account-lib/src/coin/dot/anonymousAddressInitializationBuilder.ts

```diff
@@ -0,0 +1,136 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { methods } from '@substrate/txwrapper-polkadot';
+import BigNumber from 'bignumber.js';
+import { InvalidTransactionError } from '../baseCoin/errors';
+import { TransactionBuilder } from './transactionBuilder';
+import { Transaction } from './transaction';
+import { DecodedSignedTx, DecodedSigningPayload, UnsignedTransaction } from '@substrate/txwrapper-core';
+import { TransactionType } from '../baseCoin';
+import { AddAnonymousProxyArgs, MethodNames, ProxyType } from './iface';
+import { AnonymousAddressInitializationSchema } from './txnSchema';
+
+export class AnonymousAddressInitializationBuilder extends TransactionBuilder {
```

Comment:
Is this an extension on the `AddressInitializationBuilder` ? If so, we can extend on that builder as well as the data interfaces

Reply:
We can probably modify `AddressInitializationBuilder` to do this by default. We don't really use the proxy addresses it builds as it is implemented right now.

Reply:
I added the anonymous proxy fields as a part of the origin AddressInitialization builder. This means that both addProxy and anonymous proxy transactions would use this builder - the only difference is for `addProxy`, the `owner()` function needs to be called to pass the proxy address in, and for `anonymous` the `index()` can be called to set the index (but will be defaulted to 0). Both transactions will need `delay()` and `type()`